cpu/cc26x0/cpu.c

```diff
@@ -52,11 +49,153 @@ void cpu_init(void)
     periph_init();
 }
 
-static void cpu_clock_init(void)
+//*****************************************************************************
+//
+// Disable all external interrupts
+//
+//*****************************************************************************
+#if defined(DOXYGEN)
+uint32_t
+CPUcpsid(void)
+{
+    // This function is written in assembly. See cpu.c for compiler specific implementation.
+}
+#else
+uint32_t __attribute__((naked))
+CPUcpsid(void)
+{
+    uint32_t ui32Ret;
+
+    // Read PRIMASK and disable interrupts
+    __asm volatile ("    mrs     %0, PRIMASK\n"
+                    "    cpsid   i\n"
+                    "    bx      lr\n"
+                    : "=r"(ui32Ret)
+                   );
+
+    // The return is handled in the inline assembly, but the compiler will
+    // still complain if there is not an explicit return here (despite the fact
+    // that this does not result in any code being produced because of the
+    // naked attribute).
+    return(ui32Ret);
+}
+#endif
```

Comment:
This and `CPUcpsie` can be replaced on vendor code :wink:

`s/CPUcpsid/disable_irq/`

and

`s/CPIcpsie/irq_restore/g` passing the value that `irq_disable` returned as the paremeter to check for nested `irq_disable` blocks of code.

Reply:
I don't understand, exactly how I would replace these functions, you mean the functions in `cpu/cotexm_common`?

Reply:
you would use `irq_disable()`/`irq_restore()`, no need to duplicate that functionality.